eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -476,6 +461,39 @@ public void testDifferentCacheFilesPerKey() {
         assertFalse(apiKey1Assignment);
     }
 
+    @Test
+    public void testFetchCompletesBeforeCacheLoad() {
```

Comment:
This is the test that reproduced the issue (and after the fix, verified it is handled correctly) 💪